api/docker-compose.yaml

```diff
@@ -20,165 +26,52 @@ services:
     ports:
       - 127.0.0.1:3306:3306
   api:
+    <<: *api-defaults
     depends_on:
       - db
-    build:
-      context: ./src/dev/server
-    image: allofustest/workbench-dev-api:local
-    user: ${UID}
-    working_dir: /w/api
-    volumes:
-      - src-sync:/w:nocopy
-      - gradle-cache:/.gradle
-      - ~/.config:/.config:cached
-      - ~/.gsutil:/.gsutil:cached
     command: ./project.rb start-api-and-incremental-build
-    environment:
-      - GOOGLE_APPLICATION_CREDENTIALS=/w/api/sa-key.json
-    env_file:
-      - db/vars.env
     ports:
       - 127.0.0.1:8081:8081
```

Comment:
Maybe make a named constant for the two different port numbers 8081 and 8001. If order is arbitrary, maybe list them alphabetically.

Reply:
Looks like the way to do this in docker-compose would be to push the ports into a `.env` file. local constants/variables don't seem to be possible in docker-compose, except maybe by further abusing anchors.

AFAICT the suggestion isn't really related to the PR, so I don't intend to make a non-trivial change like that here.